syllabus/3-PVM-Polkadot-Architecture-and-Smart-Contracts/5-History_of_Contracts_on_Polkadot-slides.md

```diff
@@ -0,0 +1,102 @@
+---
+title: The history of contracts on Polkadot
+description: Historical evolution of smart contracts and runtime modules (pallets) on Polkadot, including Frontier, ink!, and pallet-revive.
+duration: 15 minutes
+---
+
+# dAppChains vs. Contracts
+
+Notes:
+
+Intro  
+We'll trace the historical evolution of execution environments on Polkadot, focusing explicitly on smart contract and pallet development. Specifically: Frontier (initial EVM compatibility), ink! (WASM-based contracts), and recent developments like pallet-revive (RISC-V compatible VM with Solidity compilation).
+
+---
+
+# Early Polkadot Execution Models
+
+Initially, Polkadot had two primary approaches to application execution:
+
+- **Runtime modules (pallets)**:
+  - Native Substrate-based execution.
+  - Maximum efficiency, full runtime customization.
+  - Integrated directly into parachain runtime.
+  - Rust, steep developer learning curve.
+
+- **Smart contracts (ink!, WASM-based)**:
+  - Flexible, easier developer onboarding compared to pallets.
+  - Permissionless deployments without runtime upgrades.
+  - Limited initial adoption: lacked mature tooling, smaller developer community compared to Solidity.
+
+Notes:
+
+---
+
+# Frontier: Naive Solidity Compatibility
+
+- A Substrate pallet providing EVM compatibility.
+- Enabled direct deployment of Solidity smart contracts on Substrate-based chains (e.g., Moonbeam).
+- Immediately leveraged existing Ethereum developer tooling (MetaMask, Remix, Truffle).
+- Dramatically accelerated adoption by tapping into Solidity's mature ecosystem.
+
+Limitations:
+
+- Frontier initially naïve: straightforward EVM compatibility layer, limited deeper runtime integration.
+- Efficiency less than native pallets or pure WASM execution.
+
+Notes:
+
+---
+
+# ink!: WASM Smart Contracts
+
+ink! aimed to provide native WASM smart contracts on Polkadot/Substrate:
+
+- Technically powerful: WASM flexibility, theoretically superior performance.
+- Designed explicitly for Substrate: deep runtime integration possible.
+
+Adoption limited in practice:
+
+- Ethereum’s Solidity community strongly entrenched.
+- Limited initial tooling, smaller community engagement than Solidity.
+- Despite technical advantages, ink! struggled with widespread real-world adoption.
+
+Notes:
```

Comment:
Would it be worth mentioning the upcoming RISC-V compatible version: [ink! v6.0](https://use.ink/6.x/)?
While its alpha version is not yet available, it helps to highlight the various efforts within the ecosystem to demonstrate how this VM could be the one to rule them all. Heterogeneous sources being executed by the same VM in the same  state space.

Reply:
FWIW it's mentioned that PVM is a general purpose VM supporting other languages like Rust in the chapter about PVM.

Reply:
Just glanced at the slide, I think it should mention somewhere that ink! in an upcoming version does support PVM. But that's besides the point - the point, and a very important point for the whole module, is that a future ink! version does support Solidity ABI and Solidity interfaces (like any other language not wanting to isolate itself).

Reply:
Agreed! You are 100% correct that highlighting the Solidity ABI + interfaces support is key when discussing the language features.